packages/gestalt-codemods/40.0.0/rename-story-pin-icon-to-idea-pin.js

```diff
@@ -0,0 +1,100 @@
+/*
+ * Converts
+ *  <Icon icon='story-pin'/> to <Icon icon='idea-pin'/>
+ *  <IconButton icon='story-pin'/> to <IconButton icon='idea-pin'/>
+ *  <Pog icon='story-pin'/> to <Pog icon='idea-pin'/>
+ *  <Button iconEnd='story-pin'/> to <Button iconEnd='idea-pin'/>
+ *
+ *  Console logs each Module for a manual review.
+ */
+
+// yarn codemod --parser=flow -t=packages/gestalt-codemods/19.0.0/rename-story-pin-icon-to-idea-pin.js relative/path/to/your/code
+
+export default function transformer(file, api) {
+  const j = api.jscodeshift;
+  const src = j(file.source);
+  let localIdentifierName;
+  let localModuleIdentifierName;
+
+  let fileHasModifications = false;
+
+  src.find(j.ImportDeclaration).forEach((path) => {
+    const decl = path.node;
+    if (decl.source.value !== 'gestalt') {
+      return null;
+    }
+
+    localIdentifierName = decl.specifiers
+      .filter((node) => ['Icon', 'IconButton', 'Pog', 'Button'].includes(node.imported.name))
+      .map((node) => node.local.name);
+
+    localModuleIdentifierName = decl.specifiers
+      .filter((node) => ['Module'].includes(node.imported.name))
+      .map((node) => node.local.name);
+
+    return null;
+  });
+
+  if (!localIdentifierName && !localModuleIdentifierName) {
+    return null;
+  }
+
+  const transform = src
+    .find(j.JSXElement)
+    .forEach((jsxElement) => {
+      const { node } = jsxElement;
+
+      if (
+        !localIdentifierName.includes(node.openingElement.name.name) &&
+        !localModuleIdentifierName.includes(node.openingElement.name.name)
+      ) {
+        return null;
+      }
+
+      const attrs = node.openingElement.attributes;
+
+      if (attrs.some((attr) => attr.type === 'JSXSpreadAttribute')) {
+        throw new Error(
+          `Remove Dynamic Text properties and rerun codemod. Location: ${file.path} @line: ${node.loc.start.line}`,
```

Comment:
I don't think we're dealing with Text here 😅

Reply:
DONE